ibis/expr/operations/relations.py

```diff
@@ -78,6 +78,67 @@ def resolve_name(self):
         return self.name
 
 
+@public
+class LocalTable(PhysicalTable):
+    schema = rlz.instance_of(sch.Schema)
+    name = rlz.instance_of(str)
+
+
+@public
+class LocalWrite(TableNode, sch.HasSchema):
+    table = rlz.table
+    name = rlz.instance_of(str)
+
+    @cached_property
+    def schema(self):
+        return self.table.schema()
+
+
+@public
+class AsOfMerge(TableNode, sch.HasSchema):
```

Comment:
It's hard to tell why this has a bunch of hardcoded fields with their index as part of their name.

Why doesn't the existing `AsOfJoin` work for you?

Reply:
Not related to UDT; please skip.